acme/crypto.go

```diff
@@ -258,9 +261,22 @@ func pemDecodeTox509(pem []byte) (*x509.Certificate, error) {
 		return nil, err
 	}
 
+	if pemBlock.Type != "CERTIFICATE REQUEST" {
```

Comment:
This block is in the wrong function, isn't it? Should be in `pemDecodeTox509CSR`.